github.go

```diff
@@ -149,10 +180,34 @@ func (g *GithubLocator) ListReleases(ctx context.Context, amount int) ([]Release
 	return releases, nil
 }
 
+// GitHubLocatorOption is an option to configure GithubLocator.
+type GitHubLocatorOption interface {
+	applyGitHubLocatorOption(l *GithubLocator)
+}
```

Comment:
It scales better when we have other structs that could reuse the options. In that case, `WithTracerProvider()` implements multiple option interfaces rather than having multiple functions like `WithTracerProviderForGitHubLocator()` or `WithTracerProviderForXYZ()`

It's a common practice in well-known libraries, for example `grpc`, `otel`, etc

However, if it's a blocker, feel free to tell me, I will change it to function.

Reply:
just curiosity, not a blocker :slightly_smiling_face: